rest/server_context.go

```diff
@@ -259,15 +259,13 @@ func (sc *ServerContext) PostUpgrade(preview bool) (postUpgradeResults PostUpgra
 
 	for name, database := range sc.databases_ {
 		// View cleanup
-		removedDDocs, err := database.RemoveObsoleteDesignDocs(preview)
-		if err != nil {
-			return nil, err
-		}
+		removedDDocs, _ := database.RemoveObsoleteDesignDocs(preview)
 
 		// Index cleanup
-		removedIndexes, err := database.RemoveObsoleteIndexes(preview)
-		if err != nil {
-			return nil, err
+		var removedIndexes = make([]string, 0)
+		gocbBucket, ok := base.AsGoCBBucket(database.Bucket)
+		if ok && gocbBucket.HasN1qlNodes() {
+			removedIndexes, _ = database.RemoveObsoleteIndexes(preview)
```

Comment:
Would be nicer if this logic was pushed down into removeObsoleteIndexes (which has the advantage of already doing the AsGoCBBucket cast)